### PR TITLE
Typed print ops

### LINKED ISSUE #114: Print ops
* Typecheck `print` and `println` to expect single Text argument.
* Add `print_int`, `println_int` opcodes.
* Add `println_list_joined_using` opcode expecting a list and text.
* Add `println_many_joined_using` opcode expecting a arbitrary many arguments and then one text argument
* Add related plugins.
* Remove typecheck for print in Python.
* Add `print` and `println` to UnaryOpCode array & type.

### PR DESCRIPTION
**Pull request by MichalMarsalek**
Closes #114

### CODE REVIEW

File: src/common/getType.ts [hunk 1]
```diff
@@ -275,8 +275,11 @@ function getOpCodeType(
   program: Program
 ): Type {
   const types = getArgs(expr).map((x) => getType(x, program));
-  function expectVariadicType(expected: Type) {
-    if (types.length < 2 || types.some((x, i) => !isSubtype(x, expected))) {
+  function expectVariadicType(expected: Type, minArity = 2) {
```
Comment (MeWhenI) [inline src/common/getType.ts:278]: why default minArity to 2?

**Comment by MeWhenI** [comment]
Looks good, do you want to the joining print operators implemented before pulling?

File: src/common/getType.ts (hunk 1 above)
Comment (MichalMarsalek) [reply src/common/getType.ts:278]: > why default minArity to 2?

That was the previous behaviour. When I add new parameters to a function I tend to add a default value that keeps the old semantics.

**Comment by MichalMarsalek** [comment]
> Looks good, do you want the joining print operators implemented before pulling?

I don't.

**Pull request merged**